xmodule/course_block.py

```diff
@@ -57,6 +58,18 @@
 COURSE_VIDEO_SHARING_PER_VIDEO = 'per-video'
 COURSE_VIDEO_SHARING_ALL_VIDEOS = 'all-on'
 COURSE_VIDEO_SHARING_NONE = 'all-off'
+# .. toggle_name: FEATURES['CREATE_COURSE_WITH_DEFAULT_ENROLLMENT_START_DATE']
+# .. toggle_implementation: SettingDictToggle
+# .. toggle_default: False
+# .. toggle_description: The default behavior, when this is disabled, is that newly created course has no
+#   enrollment_start date set. When the feature is enabled - the newly created courses will have the
+#   enrollment_start_date set to DEFAULT_START_DATE.
```

Comment:
1. I fixed a typo and moved the the sentence about it being a permanent option to the description.
```suggestion
# .. toggle_description: The default behavior, when this is disabled, is that a newly created course has no
#   enrollment_start date set. When the feature is enabled - the newly created courses will have the
#   enrollment_start_date set to DEFAULT_START_DATE. This is intended to be a permanent option.
```

2. My understanding is that this toggle was added because some orgs (2U) did not want this toggled on. I do not know much about this, and if I were an operator, I would wonder why I should choose one option over the option:
  i. What is the consequence of having no start date set? Without knowing anything, that seems like it would be bad, but somehow it works for 2U. Why?
  ii. I think I read that DEFAULT_START_DATE is intended to be a date in the distant future so courses will not appear. Is that accurate, or only if you are using this toggle? Is the default used for other things? Maybe you could speak to that a bit more here?

Please update my first point, because it is simple. Regarding my second point, I am not the owner of this particular area of the code, so this is just a suggestion. You can decide how much you update the toggle docs, but I personally would rather longer docs that help me understand more completely what it is really for. That said, if you think most operators would know all of these things, maybe you've provided enough.

Reply:
Thanks for the detailed review

> What is the consequence of having no start date set?

The toggle has no effect on the **start date** but only on the **enrollment start date**, but I've got your point, I'll think about how to describe the toggle behavior with more details

> I think I read that DEFAULT_START_DATE is intended to be a date in the distant future so courses will not appear. Is that accurate, or only if you are using this toggle? Is the default used for other things? Maybe you could speak to that a bit more here?

I just reused the DEFAULT_START_DATE constant. It was originally used only as a default value for the **course start** date and no changes were made there